extensions/steve/steve.go

```diff
@@ -0,0 +1,71 @@
+package steve
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/sirupsen/logrus"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	notFound = "not found"
+)
+
+// DeleteSteveResource accepts a client, resource type and v1.SteveAPIObject and deletes the resource
+func DeleteSteveResource(client *rancher.Client, steveResourceType string, resource *v1.SteveAPIObject) error {
+	logrus.Info("Deleting resource...")
+	err := client.Steve.SteveType(steveResourceType).Delete(resource)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// VerifyDeletedSteveResource accepts a client, resource type and v1.SteveAPIObject and verifies that
+// the resource was successfully removed from the cluster
+func VerifyDeletedSteveResource(client *rancher.Client, steveResourceType string, deletedResource *v1.SteveAPIObject) error {
+	logrus.Info("Verifying resource was successfully removed...")
+	resource, err := client.Steve.SteveType(steveResourceType).ByID(deletedResource.Resource.ID)
+	if err != nil {
+		if strings.Contains(err.Error(), notFound) {
+			logrus.Info("Resource was successfully removed!")
+			return nil
+		} else {
+			return err
+		}
+	}
+
+	if resource != nil {
+		return errors.New("Resource was not removed")
+	}
+
+	return nil
+}
+
+// WaitForSteveResourceDeletion accepts a client, steve resource type, and resource ID, then waits for a steve resource to be deleted
+func WaitForSteveResourceDeletion(client *rancher.Client, interval, timeout time.Duration, steveResourceType, steveID string) error {
```

Comment:
My concern about this wait is that it's too generic and deleting resources is different depending on what that is.

Reply:
too generic? 🤔 -- could you elaborate on your concern?  When would this not be applicable for awaiting a steve resource to be deleted?  I'm not understanding what you'd like to see in place of this 🙂

Reply:
Deleting a secret is different than deleting a cluster. We would want to create extensions to help validate those differently. Having such a generic function causes confusion for someone  writing a test

Reply:
Have an extensions that waits for a node to be deleted. Why did you decide to go this route than what you had before?

Reply:
If this covers a good chunk of the "wait for delete" use cases though wouldn't it make sense to have a generic function and then if you need something more specific you could just create it in place? @igomez06

Reply:
Thanks for the clarification Israel!

this function just awaits the resource to be deleted, which would be applicable to secrets too.  Validating the secret was properly deleted should be a separate extension IMO.  This is purely to just await the resource deletion, and I tried to name the function appropriately, as to not cause any confusion.

I went this route as requested by reviewers.  The concern was using v3 vs v1 and the ask was to refactor out all the v3 calls.  This came about as the result of that re-work.  

If I edit this code to be less generic, we will have to add more extensions, extremely similar to each other, for each specific use case, which just seems unnecessary when it could be tackled through one extension.

Reply:
You make a good point though Sam, I'll drop it.